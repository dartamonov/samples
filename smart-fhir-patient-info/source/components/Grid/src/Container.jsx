import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

class Container extends Component {

  render() {
    const classes = 'container';

    const {
      children,
      className
    } = this.props;

    return (
      <div className={classNames(className, classes)}>
        {children}
      </div>
    );
  }
}

Container.propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string
};

export default Container;
