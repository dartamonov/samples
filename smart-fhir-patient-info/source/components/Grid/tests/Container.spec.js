import React from 'react';
import { create } from 'react-test-renderer';
import Container from '../src/Container.jsx';

describe("Container component", () => {
  it("requires a child component", () => {
    const tree = create(
      <Container><span>Test</span></Container>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
