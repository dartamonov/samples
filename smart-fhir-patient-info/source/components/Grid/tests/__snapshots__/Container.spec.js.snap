// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component requires a child component 1`] = `
<div
  className="container"
>
  <span>
    Test
  </span>
</div>
`;
